flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -240,14 +234,13 @@ public Task(TaskDeploymentDescriptor tdd,
 		checkArgument(tdd.getNumberOfSubtasks() > 0);
 		checkArgument(tdd.getIndexInSubtaskGroup() >= 0);
 		checkArgument(tdd.getIndexInSubtaskGroup() < tdd.getNumberOfSubtasks());
+		checkArgument(tdd.getAttemptNumber() >= 0);
```

Comment:
You can probably skip these checks, given that `TaskInfo` has these checks already...